test/e2e/case32_secret_stringdata_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"open-cluster-management.io/config-policy-controller/test/utils"
+)
+
+const (
+	case32ConfigPolicyName string = "case32config"
+	case32CreatePolicyYaml string = "../resources/case32_secret_stringdata/case32_create_secret.yaml"
+)
+
+var _ = Describe("Test converted stringData being decoded before comparison for Secrets", Ordered, func() {
+	It("Config should be created properly on the managed cluster", func() {
+		By("Creating " + case32ConfigPolicyName + " on managed")
+		utils.Kubectl("apply", "-f", case32CreatePolicyYaml, "-n", testNamespace)
+		cfg := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+			case32ConfigPolicyName, testNamespace, true, defaultTimeoutSeconds)
+		Expect(cfg).NotTo(BeNil())
+	})
+
+	It("Verifies the config policy is initially compliant "+case32ConfigPolicyName+" in "+testNamespace, func() {
+		By("Waiting for " + case32ConfigPolicyName + " to become Compliant")
+		Eventually(func() interface{} {
+			cfgplc := utils.GetWithTimeout(
+				clientManagedDynamic, gvrConfigPolicy,
+				case32ConfigPolicyName, testNamespace,
+				true, defaultTimeoutSeconds,
+			)
+
+			return utils.GetComplianceState(cfgplc)
+		}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+	})
+
+	It("Verifies that a secret is created by "+case32ConfigPolicyName+" in openshift-config", func() {
+		By("Grabbing htpasswd-secret from namespace openshift-config")
+		scrt := utils.GetWithTimeout(
+			clientManagedDynamic, gvrSecret, "htpasswd-secret", testNamespace, true, defaultTimeoutSeconds,
+		)
+
+		Expect(scrt).NotTo(BeNil())
+	})
+
+	It("Verifies the config policy stays compliant "+case32ConfigPolicyName+" in "+testNamespace, func() {
+		By("Checking the events on the configuration policy" + case32ConfigPolicyName)
+		Consistently(func() int {
+			eventlen := len(utils.GetMatchingEvents(
+				clientManaged, testNamespace,
+				case32ConfigPolicyName, case32ConfigPolicyName,
+				"NonCompliant;", defaultTimeoutSeconds))
+
+			return eventlen
+		}, defaultTimeoutSeconds, 5).Should(BeNumerically("<", 3))
```

Comment:
We need to increase consistently checking the time, cuz I tested with an unchanged(just main branch build) controller accidentally.. it passed

Reply:
Also the screenshot in the ticket, flapping happened every **one min**

Reply:
I suggest ```Should(Equal(1))``` to more accurate test

Reply:
This ended up being a bigger issue than I initially thought - turns out that the old code and the new code had the same number of events containing `NonCompliant;` as well as `Compliant;`, but the messages were slightly different. The old one would say in the message that the Secret is being "updated", while the new (correct) code has an event with a message that states that there are no actions to be taken. Thus, I modified the `Consistently` block to check for the `updated` keyword in the message instead of the original `NonCompliant` keyword.

Also, I ended up adding the evaluationInterval back in the yaml. Manual testing showed that by default, the policy was not getting evaluated often enough (around every minute or so) which was increasing the testing period.

Reply:
If only adding evaluationInterval works.... this is another issue.. can you raise this in our standup?

Reply:
Also the messages should present`NonCompliant` or `Compliant`